File: solutions/code/1. Two Sum/solution2.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> hashmap;

        for (int i = 0; i < nums.size(); i++) {
            int num = nums[i];
            if (hashmap.count(target - num) > 0) {
                return {hashmap[target - num], i};
            }

            hashmap[num] = i;
        }

        return {-1, -1};
    }
};

File: solutions/code/0001. Two Sum/solution2.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int n = nums.length;
        Map<Integer, Integer> hashmap = new HashMap<>();

        for (int i = 0; i < n; i++) {
            int num = nums[i];
            if (hashmap.containsKey(target - num)) {
                return new int[] {hashmap.get(target - num), i};
            }
            hashmap.put(num, i);
        }


        return null;
    }
}